src/Methods/InvoicePaymentMethod.php

```diff
@@ -262,37 +207,63 @@ public function getDescription():string
      */
     public function isSwitchableTo(int $orderId = null):bool
     {
+        /** @var InvoiceLimitationsService $helper */
+        $helper = pluginApp(InvoiceLimitationsService::class);
+        
+        //  If order ID is given check the order data
         if(!is_null($orderId) && $orderId > 0) {
             /** @var OrderRepositoryContract $orderRepo */
             $orderRepo = pluginApp(OrderRepositoryContract::class);
             $filters = $orderRepo->getFilters();
             $filters['addOrderItems'] = false;
             $orderRepo->setFilters($filters);
+            
             try {
-                $order = $orderRepo->findOrderById($orderId, ['amounts']);
-                $amount = $order->amount;
-                
-                /** @var CurrencyExchangeRepository $currencyService */
-                $currencyService = pluginApp(CurrencyExchangeRepositoryContract::class);
-                $minAmount = (float)$currencyService->convertFromDefaultCurrency($amount->currency, $this->settings->getSetting('minimumAmount'));
-                $maxAmount = (float)$currencyService->convertFromDefaultCurrency($amount->currency, $this->settings->getSetting('maximumAmount'));
+                $order = $orderRepo->findOrderById($orderId, ['amounts', 'addresses']);
+                $contact = $order->contactReceiver;
                 
-                /**
-                 * Check the minimum amount
-                 */
-                if( $minAmount > 0.00 && $amount->invoiceTotal < $minAmount) {
-                    return false;
-                }
-        
-                /**
-                 * Check the maximum amount
-                 */
-                if( $maxAmount > 0.00 && $maxAmount < $amount->invoiceTotal) {
-                    return false;
-                }
-            } catch(\Exception $e) {}
+                return $helper->respectsAllLimitations(
+                    pluginApp(SettingsHelper::class, [$this->settings, $order->plentyId]),
+                    $contact === null,
```

Comment:
Isn't there a second attribute to check, wheither or not it deals with a guest order. The Contact-Model contains the following Method:
```php
	/**
	 * Accessor for "access type" contact option
	 *
	 * @internal
	 * @return string 1 if access type is guest, otherwise empty string
	 */
	public function getSingleAccessAttribute()
	{
		$option = $this->findOption(ContactOption::TYPE_ACCESS, ContactOption::SUBTYPE_GUEST);

		if ($option != null)
		{
			return $option->value;
		}
		return "";
	}
```
This seems to me, that an additional should be made in order to verify if an order has been placed by a guest, like so:
```php
$isGuest = !($contact === null || $contact->getSingleAccess == 1);
``` 
Or am I wrong here?